src/porepy/viz/exporter.py

```diff
@@ -520,546 +1064,789 @@ def _export_grid(
+                if cell_type not in cell_to_faces:
+                    cell_to_faces[cell_type] = []
+                cell_to_faces[cell_type] += [
+                    [
+                        fn_indices[fn_indptr[f] : fn_indptr[f + 1]]  # nodes
+                        for f in cf_indices[cf_indptr[c] : cf_indptr[c + 1]]  # faces
+                    ]
+                    for c in cells  # cells
+                ]
 
-            pts_pos += g.num_nodes
+            # Update offset
+            nodes_offset += g.num_nodes
+            cell_offset += g.num_cells
 
-        # NOTE: only cell->faces relation will be kept, so far we export only polyhedron
-        # otherwise in the next lines we should consider also the cell->nodes map
+        # Determine the total number of blocks / cell types.
+        num_blocks = len(cell_to_faces)
 
-        # construct the meshio data structure
-        num_block = len(cell_to_nodes)
-        meshio_cells = np.empty(num_block, dtype=object)
-        meshio_cell_id = np.empty(num_block, dtype=object)
+        # Initialize the meshio data structure for the connectivity and cell ids.
+        meshio_cells = np.empty(num_blocks, dtype=object)
+        meshio_cell_id = np.empty(num_blocks, dtype=object)
 
+        # Store the cells in meshio format
         for block, (cell_type, cell_block) in enumerate(cell_to_faces.items()):
+            # Adapt the block number taking into account of previous cell types.
             meshio_cells[block] = meshio.CellBlock(cell_type, cell_block)
             meshio_cell_id[block] = np.array(cell_id[cell_type])
 
-        return meshio_pts, meshio_cells, meshio_cell_id
+        # Return final meshio data: points, cell (connectivity), cell ids
+        return Meshio_Geom(meshio_pts, meshio_cells, meshio_cell_id)
 
-    def _write(self, fields: Iterable[Field], file_name: str, meshio_geom) -> None:
+    def _write(
+        self,
+        fields: Iterable[Field],
+        file_name: str,
+        meshio_geom: Meshio_Geom,
+    ) -> None:
+        """
+        Interface to meshio for exporting cell data.
 
+        Parameters:
+            fields (iterable, Field): fields which shall be exported
+            file_name (str): name of final file of export
+            meshio_geom (Meshio_Geom): Namedtuple of points,
+                connectivity information, and cell ids in
+                meshio format (for a single dimension).
+        """
+        # Initialize empty cell data dictinary
         cell_data = {}
 
-        # we need to split the data for each group of geometrically uniform cells
-        cell_id = meshio_geom[2]
-        num_block = cell_id.size
-
+        # Split the data for each group of geometrically uniform cells
+        # Utilize meshio_geom for this.
         for field in fields:
+
+            # TODO RM? as implemented now, field.values should never be None.
             if field.values is None:
```

Comment:
In that case, I suggest
```suggestion
           assert field.values is not None
```